bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/flamingos2/Flamingos2Recipe.java

```diff
@@ -67,6 +75,162 @@ public ItcSpectroscopyResult serviceResult(final SpectroscopyResult r, final boo
     }
 
     public SpectroscopyResult calculateSpectroscopy() {
+        final CalculationMethod calcMethod = _obsDetailParameters.calculationMethod();
+        Log.fine("calcMethod = " + calcMethod);
+
+        if (calcMethod instanceof SpectroscopyS2N) {
+            numberExposures = ((SpectroscopyS2N) calcMethod).exposures();
+        } else if (calcMethod instanceof SpectroscopyInt) {
+            numberExposures = ((SpectroscopyInt) calcMethod).exposures();
+        } else {
+            throw new Error("Unsupported calculation method");
+        }
+        Log.fine("numberExposures = " + numberExposures);
+
+        final double skyAper = 1.0;  // hard-coded in the F2 web form
+        double readNoise;
+        double totalTime;
+
+        if (calcMethod instanceof SpectroscopyInt) {
+            // Determine exposureTime & numberExposures that will give the requested S/N at wavelength.
+
+            double desiredSNR = ((SpectroscopyInt) calcMethod).sigma();
+            Log.fine(String.format("desiredSNR = %.2f", desiredSNR));
+
+            double wavelength = ((SpectroscopyInt) _obsDetailParameters.calculationMethod()).wavelength();
+            Log.fine(String.format("Wavelength = %.2f nm", wavelength));
+
+            double maxFlux = instrument.maxFlux();  // maximum useful flux
+            Log.fine(String.format("maxFlux = %.0f e-", maxFlux));
+
+            // Perform an initial run of the ITC to get the signal and background for this target at this wavelength
+            readMode = ReadMode.MEDIUM_OBJECT_SPEC;  // arbitrary choice
```

Comment:
Will this be a parmeter in the future?

Reply:
We may be able to avoid it by setting the read mode based on the returned exposure time.